src/app/clusters/scenes/SceneTable.h

```diff
@@ -0,0 +1,363 @@
+        {
+            mEndpointId = kInvalidEndpointId;
+            mGroupId    = kGlobalGroupSceneId;
+            mSceneId    = kUndefinedSceneId;
+        }
+        bool operator==(const SceneStorageId & other)
+        {
+            return (this->mEndpointId == other.mEndpointId && this->mGroupId == other.mGroupId && this->mSceneId == other.mSceneId);
+        }
+        void operator=(const SceneStorageId & other)
+        {
+            this->mEndpointId = other.mEndpointId;
+            this->mGroupId    = other.mGroupId;
+            this->mSceneId    = other.mSceneId;
+        }
+    };
+
+    /// @brief struct used to store data held in a scene
+    struct SceneData
+    {
+        char mName[kSceneNameMax]                = { 0 };
+        size_t mNameLength                       = 0;
+        SceneTransitionTime mSceneTransitionTime = 0;
+        ExtensionFieldSetsImpl mExtensionFieldSets;
+        TransitionTime100ms mTransitionTime100ms = 0;
+        CharSpan mNameSpan;
+
+        SceneData(const CharSpan & sceneName = CharSpan(), SceneTransitionTime time = 0, TransitionTime100ms time100ms = 0) :
+            mSceneTransitionTime(time), mTransitionTime100ms(time100ms)
+        {
+            this->SetName(sceneName);
+        }
+        SceneData(ExtensionFieldSetsImpl fields, const CharSpan & sceneName = CharSpan(), SceneTransitionTime time = 0,
+                  TransitionTime100ms time100ms = 0) :
+            mSceneTransitionTime(time),
+            mTransitionTime100ms(time100ms)
+        {
+            this->SetName(sceneName);
+            mExtensionFieldSets = fields;
+        }
+        SceneData(const SceneData & other) :
+            mSceneTransitionTime(other.mSceneTransitionTime), mTransitionTime100ms(other.mTransitionTime100ms)
+        {
+            this->SetName(other.mNameSpan);
+            mExtensionFieldSets = other.mExtensionFieldSets;
+        }
+        ~SceneData(){};
+
+        CHIP_ERROR Serialize(TLV::TLVWriter & writer) const
+        {
+            TLV::TLVType container;
+            ReturnErrorOnFailure(
+                writer.StartContainer(TLV::ContextTag(kTagSceneDataContainer), TLV::kTLVType_Structure, container));
+
+            // A 0 size means the name wasn't used so it won't get stored
+            if (!this->mNameSpan.empty())
+            {
+                ReturnErrorOnFailure(writer.PutString(TLV::ContextTag(kTagSceneName), this->mNameSpan));
+            }
+
+            ReturnErrorOnFailure(
+                writer.Put(TLV::ContextTag(kTagSceneDTransitionTime), static_cast<uint16_t>(this->mSceneTransitionTime)));
+            ReturnErrorOnFailure(
+                writer.Put(TLV::ContextTag(kTagSceneDTransitionTime100), static_cast<uint8_t>(this->mTransitionTime100ms)));
+            ReturnErrorOnFailure(this->mExtensionFieldSets.Serialize(writer));
+
+            return writer.EndContainer(container);
+        }
+        CHIP_ERROR Deserialize(TLV::TLVReader & reader)
+        {

```

Comment:
Why are these implementation details in the interface, and not in SceneTableImpl.cpp ?

Reply:
Moved to implementation